### PR TITLE
feat: persist tracking value in storage

### PR DESCRIPTION
**Pull request by andrzej-stencel**

### CODE REVIEW

File: receiver/sqlqueryreceiver/logs_receiver.go [hunk 1]
```diff
@@ -259,5 +306,9 @@ func rowToLog(row stringMap, config LogsCfg, logRecord plog.LogRecord) error {
 }
 
 func (queryReceiver *logsQueryReceiver) shutdown(ctx context.Context) error {
-	return nil
+	var errors error
+	// if queryReceiver.db != nil {
```
Comment (kasia-kujawa) [inline receiver/sqlqueryreceiver/logs_receiver.go]: Why is this here in this form? I mean the commented code.

File: receiver/sqlqueryreceiver/README.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ The configuration supports the following top-level fields:
   e.g. _host=localhost port=5432 user=me password=s3cr3t sslmode=disable_
 - `queries`(required): A list of queries, where a query is a sql statement and one or more `logs` and/or `metrics` sections (details below).
 - `collection_interval`(optional): The time interval between query executions. Defaults to _10s_.
+- `storage` (optional, default `""`): The ID of a storage extension to be used to [track processed results](#tracking-processed-results).
```
Comment (kasia-kujawa) [inline receiver/sqlqueryreceiver/README.md]: please add here link to the documentation of storage extension

Comment (andrzej-stencel) [reply receiver/sqlqueryreceiver/README.md]: Added link to the File Storage extension. I couldn't find general documentation about storage extensions.

File: receiver/sqlqueryreceiver/logs_receiver.go (hunk 1 above)
Comment (andrzej-stencel) [reply receiver/sqlqueryreceiver/logs_receiver.go]: Thanks, this was some testing; removed.

**Review by kasia-kujawa — APPROVED** [review]
